unmarshal.go

```diff
@@ -0,0 +1,336 @@
+package ipld
```

Comment:
Marshalling and unmarshalling are extra utilities. They shouldn't be in this PR perhaps but included elsewhere.